services/core/users/src/main/java/org/lfenergy/operatorfabric/users/controllers/UsersController.java

```diff
@@ -125,6 +125,7 @@ public UserSettings fetchUserSetting(HttpServletRequest request, HttpServletResp
     public UserSettings patchUserSettings(HttpServletRequest request, HttpServletResponse response, String login, UserSettings userSettings) throws Exception {
         UserSettingsData settings = userSettingsRepository.findById(login)
                 .orElse(UserSettingsData.builder().login(login).build());
+        publisher.publishEvent(new UpdatedUserEvent(this, busServiceMatcher.getServiceId(), login));
```

Comment:
Do it only if patching processesStatesNotNotified

Reply:
Sonar warnings solved. 
Complexity of method checkIfUserMustReceiveTheCard(...) reduced.
The karate tests will be done with another ticket.